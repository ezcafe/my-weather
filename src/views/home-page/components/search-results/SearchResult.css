.search-result {
    border: 1px solid #ccc;
    border-radius: 5px;
    margin: 0 5px;
    padding: 24px;
    text-align: center;
}

.search-result > h2 {
    font-size: 20px;
    margin-bottom: 5px;
}

.search-result > span {
    display: block;
    font-size: 16px;
}
