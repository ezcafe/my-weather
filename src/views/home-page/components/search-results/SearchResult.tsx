import { Col } from 'react-bootstrap';
import dayjs from 'dayjs';
import LocationConsolidatedWeatherModel from '../../../../stores/locations/models/LocationConsolidatedWeatherModel';
import './SearchResult.css';

const RESPONSED_DATE_FORMAT = 'YYYY-MM-DD';
const DISPLAYED_DATE_FORMAT = 'dddd';

interface iProps {
    item: LocationConsolidatedWeatherModel;
}

const SearchResult: React.FC<iProps> = ({ item }) => {
    const { applicable_date, min_temp, max_temp } = item;
    const date = dayjs(applicable_date, RESPONSED_DATE_FORMAT).format(
        DISPLAYED_DATE_FORMAT
    );
    const max = Math.floor(max_temp);
    const min = Math.ceil(min_temp);

    return (
        <Col>
            <div className="search-result">
                <h2>{date}</h2>
                <span>{`Min: ${min}`}</span>
                <span>{`Max: ${max}`}</span>
            </div>
        </Col>
    );
};

export default SearchResult;
