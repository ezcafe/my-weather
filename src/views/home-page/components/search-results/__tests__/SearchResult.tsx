import { render } from '@testing-library/react';
import SearchResult from '../SearchResult';
import { mockedLocationConsolidatedWeather } from '../../../../../fixtures';

describe('SearchResult', () => {
    test('should show component', () => {
        const firstItem = mockedLocationConsolidatedWeather;
        const { container } = render(<SearchResult item={firstItem} />);
        expect(container).toMatchSnapshot();
    });
});
