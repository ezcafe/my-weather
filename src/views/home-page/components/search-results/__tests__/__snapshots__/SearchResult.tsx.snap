// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should show component 1`] = `
<div>
  <div
    class="col"
  >
    <div
      class="search-result"
    >
      <img
        alt="Heavy Cloud"
        src="https://www.metaweather.com/static/img/weather/hc.svg"
      />
      <h2>
        Wednesday
      </h2>
      <span>
        Min: 8
      </span>
      <span>
        Max: 12
      </span>
    </div>
  </div>
</div>
`;
